test/unit/ngsiv2/amqpBinding-test.js

```diff
@@ -0,0 +1,223 @@
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        nock.cleanAll();
+
+        oldResource = config.iota.defaultResource;
+        config.iota.defaultResource = '/iot/json';
+
+        contextBrokerMock = nock('http://192.168.1.1:1026')
+            .matchHeader('fiware-service', 'smartGondor')
+            .matchHeader('fiware-servicepath', '/gardens')
+            .post('/v1/updateContext')
+            .reply(200, utils.readExampleFile('./test/contextResponses/multipleMeasuresSuccess.json'));
+
+        async.series([
+            apply(iotagentMqtt.start, config),
+            apply(request, provisionOptions),
+            apply(startConnection, config.amqp.exchange)
+        ], done);
+    });
+
+    afterEach(function(done) {
+        nock.cleanAll();
+
+        amqpConn.close();
+        config.iota.defaultResource = oldResource;
+
+        async.series([
+            iotAgentLib.clearAll,
+            iotagentMqtt.stop
+        ], done);
+    });
+
+    describe('When a new single measure arrives to a Device routing key', function() {
+        beforeEach(function() {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20MQTT%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/singleMeasureAMQP.json'))
+                .reply(204);
+        });
+
+        it('should send a new update context request to the Context Broker with just that attribute', function(done) {
+            channel.publish(config.amqp.exchange, '.1234.MQTT_2.attrs.a', new Buffer('23'));
+
+            setTimeout(function() {
+                contextBrokerMock.done();
+                done();
+            }, 100);
+        });
+    });
+
+    describe('When a new measure arrives for an unprovisioned Device', function() {
+        var groupCreation = {
+            url: 'http://localhost:4041/iot/services',
+            method: 'POST',
+            json: utils.readExampleFile('./test/groupProvisioning/provisionFullGroupAMQP.json'),
+            headers: {
+                'fiware-service': 'TestService',
+                'fiware-servicepath': '/testingPath'
+            }
+        };
+
+        beforeEach(function(done) {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v1/updateContext')
+                .reply(200, utils.readExampleFile('./test/contextResponses/multipleMeasuresSuccess.json'));
+
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v2/entities/SensorMachine:MQTT_UNPROVISIONED/attrs',
```

Comment:
AMQP_UNPROVISIONED ?

Reply:
Functionally, MQTT_UNPROVISIONED is correct because as you can see in https://github.com/dcalvoalonso/iotagent-json/blob/task/activeAttsNgsi2/test/unit/ngsiv2/amqpBinding-test.js#L149, the same name is used to publish send the update using AMQP.

Again, we have used as baseline the corresponding NGSIv1 and I imagine that this comes from old versions of the IOTA. Maybe, a correct term would be JSON_UNPROVISIONED?

Again, to be coherent, NGSIv1 should be also updated.

Reply:
I like JSON_UNPROVISIONED.

With regards to NGSIv1, I'll let them as they are now. If NGSIv1 code is transient (although the transitory period can be long ;) probably doesn't worth the time to fix also there.

Reply:
Fixed in e2760b3aee7d517d554809fed0e2f7f1a2102a13